src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy

```diff
@@ -1390,12 +1387,22 @@ public enum DarknessAblaze implements LogicCardInfo {
         bwAbility "Primal Law", {
           text "If this Pokémon is your Active Pokémon, your opponent can’t play any Pokémon from their hand to evolve their Pokémon."
           delayedA {
-            before PLAY_EVOLUTION, {
-              if (self.active && bg.currentTurn == self.owner.opposite && ef.cardToPlay.player.pbg.hand.contains(ef.cardToPlay)) {
-                wcu "$self.name's Primal Law prevents playing Pokémon from your hand to evolve your Pokémon"
+            def warnAndPrevent = {
+              if (self.active && bg.currentTurn == self.owner.opposite) {
+                wcu "$self's Primal Law prevents playing Pokémon from your hand to evolve your Pokémon"
                 prevent()
```

Comment:
Not sure this works here 🤔 `prevent()` is a function of the `before` closure's `delegate`. Have you tested this?

Reply:
Tested it yesterday and it seemed to work. Just tested with the last changes and debugging, prevent executes without issue 👍

Reply:
I have, and it seems to work fine. Pretty sure this closure's delegate is the before, then the before's delegate has a prevent, so it kind of just bubbles up the tree.